a10_octavia/controller/worker/tasks/glm_tasks.py

```diff
@@ -0,0 +1,253 @@
+                    license_net_id = CONF.a10_controller_worker.amp_boot_network_list[0]
+                else:
+                    LOG.warning("No networks were configured therefore "
+                                "nameservers cannot be set on the "
+                                "vThunder-Amphora %s", vthunder.id)
+                    return None, None
+
+        license_net = self.network_driver.get_network(license_net_id)
+        if len(license_net.subnets) < 1:
+            LOG.warning("Network %s did not have subnet configured "
+                        "therefore nameservers cannot be set.", license_net_id)
+            return None, None
+        license_subnet_id = license_net.subnets[0]
+        license_subnet = self.network_driver.show_subnet_detailed(license_subnet_id)
+
+        primary_dns = None
+        secondary_dns = None
+        subnet_dns = license_subnet.get('dns_nameservers', [])
+        if len(subnet_dns) == 1:
+            primary_dns = subnet_dns[0]
+        elif len(subnet_dns) >= 2:
+            primary_dns = subnet_dns[0]
+            secondary_dns = subnet_dns[1]
+            if len(subnet_dns) > 2:
+                LOG.warning("More than one DNS nameserver detected on subnet %s. "
+                            "Using %s as primary and %s as secondary.",
+                            license_subnet_id, primary_dns, secondary_dns)
+
+        if flavor and flavor.get('use-network-dns'):
+            if not primary_dns and not secondary_dns:
+                LOG.warning("The flavor option `use_network_dns` was defined "
+                            "but no nameservers were found on the network. "
+                            "Nameservers will not be configure on the "
+                            "vThunder-Amphora %s", vthunder.id)
+            return primary_dns, secondary_dns
+
+        if CONF.glm_license.primary_dns:
+            primary_dns = CONF.glm_license.primary_dns
+
+        if CONF.glm_license.secondary_dns:
+            secondary_dns = CONF.glm_license.secondary_dns
+
+        if flavor:
+            dns_flavor = flavor.get('dns')
+            if dns_flavor and dns_flavor.get('primary-dns'):
+                primary_dns = dns_flavor.get('primary-dns')
+            if dns_flavor and dns_flavor.get('secondary-dns'):
+                secondary_dns = dns_flavor.get('secondary-dns')
+
+        return primary_dns, secondary_dns
+
+    @axapi_client_decorator
+    def execute(self, vthunder, flavor=None):
+        if not vthunder:
+            LOG.warning("No vthunder therefore dns cannot be assigned.")
+            return None
+
+        flavor = flavor if flavor else {}
+        primary_dns, secondary_dns = self._get_dns_nameservers(vthunder, flavor)
+        if not primary_dns and secondary_dns:
+            LOG.error("A secondary DNS with IP %s was specified without a primary DNS",
+                      primary_dns)
+            raise a10_ex.PrimaryDNSMissing(secondary_dns)
```

Comment:
Can we raise only when flexpool_token is configured (for backward compatibility)?
Otherwise after upgrade the a10-octavia (or git pull) and without add dns configuration, the a10-octavia will always raise here.

Reply:
This only raises if the secondary dns is set without a primary. Even if this occurs without a token it is still not permitted.

```
        elif not primary_dns and not secondary_dns:
            return None
```
The elif handles the case where no config is set.

Reply:
ok